apps/merkle_patricia_tree/lib/merkle_patricia_tree/caching_trie.ex

```diff
@@ -66,17 +110,58 @@ defmodule MerklePatriciaTree.CachingTrie do
       remove_key(caching_trie, key)
     else
       key_nibbles = Helper.get_nibbles(key)
+      trie_change = {:update, caching_trie.in_memory_trie.root_hash, key, value}
```

Comment:
A similar thought here around extracting a function to tack onto the end of our function chain below https://github.com/mana-ethereum/mana/pull/552/files#diff-4ae4307fc5394786d95e029e0b1bb41eR125